PowerPointLabs/PowerPointLabs/NarrationsLab/Views/NarrationsLabMainSettingsPage.xaml.cs

```diff
@@ -0,0 +1,203 @@
+                else
+                {
+                    instance.voiceList.Visibility = Visibility.Visible;
+                    instance.azureVoiceBtn.Visibility = Visibility.Collapsed;
+                    instance.changeAcctBtn.Visibility = Visibility.Visible;
+                    instance.logoutBtn.Visibility = Visibility.Visible;
+                    instance.RadioAzureVoice.IsEnabled = true;
+                }
+            }
+            return instance;
+        }
+
+        public void SetNarrationsLabMainSettings(int selectedVoiceIndex, AzureVoice humanVoice, List<string> voices, bool isAzureVoiceSelected, bool isPreviewChecked)
+        {
+            voiceSelectionInput.ItemsSource = voices;
+            voiceSelectionInput.ToolTip = NarrationsLabText.SettingsVoiceSelectionInputTooltip;
+            voiceSelectionInput.Content = voices[selectedVoiceIndex];
+
+            if (humanVoice != null)
+            {
+                voiceList.SelectedItem = humanVoice;
+            }
+
+            previewCheckbox.IsChecked = isPreviewChecked;
+            previewCheckbox.ToolTip = NarrationsLabText.SettingsPreviewCheckboxTooltip;
+
+            RadioAzureVoice.IsChecked = isAzureVoiceSelected;
+            RadioDefaultVoice.IsChecked = !isAzureVoiceSelected;
+
+        }
+
+        public void Destroy()
+        {
+            instance = null;
+        }
+
+        private bool IsValidUserAccount(bool showErrorMessage = true)
+        {
+            try
+            {
+                string _key = UserAccount.GetInstance().GetKey();
+                string _endpoint = EndpointToUriMapping.regionToEndpointMapping[UserAccount.GetInstance().GetRegion()];
+                Authentication auth = Authentication.GetInstance(_endpoint, _key);
+                string accessToken = auth.GetAccessToken();
+                Console.WriteLine("Token: {0}\n", accessToken);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Failed authentication.");
+                Console.WriteLine(ex.ToString());
+                Console.WriteLine(ex.Message);
+                if (showErrorMessage)
+                {
+                    MessageBox.Show("Failed authentication");
+                }
+                UserAccount.GetInstance().Clear();
+                NarrationsLabStorageConfig.DeleteUserAccount();
+                return false;
+            }
+            return true;
+        }
+
+        private void OkButton_Click(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                string defaultVoiceSelected = RadioDefaultVoice.IsChecked == true ? voiceSelectionInput.Content.ToString() : null;
+                AzureVoice azureVoiceSelected = RadioAzureVoice.IsChecked == true ? (AzureVoice)voiceList.SelectedItem : null;
+                if (azureVoiceSelected == null && RadioAzureVoice.IsChecked == true)
+                {
+                    throw new Exception();
```

Comment:
Will there be any contextual information added?

Reply:
Edited!